proto/momento/auth.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
```

Comment:
Since this is a public repo and this is still in flux, do you think it would make sense to leave a comment at the top? Not entirely sure how much attention this code is getting, but it is going to be a minute since this is all implemented?

Reply:
If someone tries to use it they'll get a fine message back from grpc like `Error invoking method "momento.Auth/Login": target server does not expose service "momento.Auth"`. We don't currently offer support for people writing their own clients & the Auth service will soon be a fully supported service. I don't really want to add disclaimers to it <3

Reply:
Cool that makes sense. Just wanted to check.

Reply:
Yeah, it's a worthwhile thing to talk about! In general I'm okay with putting production-destined api definitions in production locations. This one is pretty well-understood and has a pattern we're laying down right now so it shouldn't be too much of an upset to have the proto definitions ahead of the implementation's availability.